exporter/prometheusexporter/accumulator.go

```diff
@@ -204,6 +209,12 @@ func (a *lastValueAccumulator) accumulateSum(metric pmetric.Metric, il pcommon.I
 			continue
```

Comment:
The line above skips when the new point's end timestamp is before the last-accumulated end timestamp.

If the new point's start timestamp is before the last-accumulated end timestamp, then there is an **overlap** condition. OTel has some guidance: https://opentelemetry.io/docs/reference/specification/metrics/datamodel/#overlap

I don't think this will be an ordinary situation, it would be OK to reset the stream when overlap occurs. So, if the arriving point's start time is < the stored end time, reset; if the arriving point's start time equals the stored end time, accumulate; if the arriving point's start time is later than the stored end time, reset.

Reply:
Yeah I was thinking the same flow, but the second case here `if the arriving point's start time equals the stored end time, accumulate` never happened on my tests, could it mean that statdreceiver is setting an incorrect start time?

Reply:
I inspected the statsd receiver code and it is behaving incorrectly. The code has a bunch of references to a `timeNowFunc()` which is set to `time.Now` except during testing. The problem is that the `timeNowFunc()` result used to set the current time is not the same as the one used to set the last-timestamp on the next round. The cleanup is to make sure that `timeNowFunc()` is called once per export and the value is used both as the current time and the next-last time.

Reply:
The above-described bug is now fixed below.